### PR TITLE
 stat.rs: Refactor to remove #[allow(clippy::cognitive_complexity)] #5881 

### PR DESCRIPTION
**Pull request by biplab5464**

### CODE REVIEW

File: src/uu/stat/src/stat.rs [hunk 1]
```diff
@@ -404,122 +404,136 @@ fn print_unsigned_hex(
...
+            format: chars[*i],
+        })
+    }
+
+    fn handle_escape_sequences(
+        chars: &[char],
+        i: &mut usize,
+        bound: usize,
+        format_str: &str,
+        use_printf: bool,
```
Comment (cakebaker) [inline src/uu/stat/src/stat.rs:481]: You can simplify this function by removing `use_printf` and handling the `false` case in `generate_tokens`.

File: src/uu/stat/src/stat.rs [hunk 2]
```diff
@@ -529,7 +543,6 @@ impl Stater {
         }
         Ok(tokens)
     }
-
```
Comment (cakebaker) [inline src/uu/stat/src/stat.rs:532]: I don't understand why you remove this empty line, it separates the two functions.

File: src/uu/stat/src/stat.rs (hunk 1 above)
Comment (biplab5464) [reply src/uu/stat/src/stat.rs]: ok, i am moving the use_printf if else logic to the generate tokens function

**Comment by cakebaker** [comment]
Good work, thanks!

**Comment by biplab5464** [comment]
Thanks for you cooperation

**Pull request merged**